### PR TITLE
Default pin state

### PR DESCRIPTION
**Pull request by andydhobbs**
Added a default state to the class constructor to allow for a defined default state.

Using the state of a switch at initialisation can lead to spurious results, for example a door switch with the door open as system restart. This could cause the behaviour to be the inverse of what is expected.

### CODE REVIEW

File: v3/primitives/pushbutton.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def __init__(self, pin, suppress=False):
         self._lf = False
         self._ld = False  # Delay_ms instance for long press
         self._dd = False  # Ditto for doubleclick
-        self.sense = pin.value()  # Convert from electrical to logical value
+        self.sense = pin.value() if sense is None else sense  # Convert from electrical to logical value
```
Comment (andydhobbs) [inline v3/primitives/pushbutton.py:29]: Considered using Pin.pull() to get the pin set up, but in many cases (mine included) the internal pull up resistors are not configured, in favour of external resistors.

**Comment by peterhinch** [comment]
The `Pushbutton` class is intended to represent a pushbutton. At power up physical pushbuttons can be assumed to be un-pressed. Hence the class was designed to deduce its logical state from its initial value, enabling the class to work for either type of contact wired to Gnd or Vcc.

The `Switch` class is intended for cases where the relationship between physical and logical state is defined in code. The only things the `Pushbutton` adds are the long press and double press events, which are unlikely to be relevant to a door closure switch. I therefore suggest you look at the `Switch` class.

**Comment by andydhobbs** [comment]
Agreed, I am actually already using the `Switch` class in some instances.  I have another example though which is perhaps less specific:  Power to the device could go through a loss/restore cycle at any point, and the user may have a button pressed at that moment.  E.g. changing the batteries in a hand held device.  If the user has a button depressed accidentally at this point they will then experience odd behaviour and a potentially hard to trace bug where a user reports randomly inverting button actions after power failure.

In my particular case, I have 3 buttons on the dashboard connected together on the reset pin of the device.  Each button functions normally in its own right (they need to support long press) but when all three are pressed the reset occurs.  As the device resets and boots up very quickly it is likely that user will still have the buttons pressed during boot up.  I can alleviate this but setting `sense` directly in my code after instantiating the class, but modifying an object property directly doesn't fit well with OOP design and is likely to break without warning in future code versions.

**Comment by peterhinch** [comment]
OK. I think your use-case is rather specialised.

I wrote these classes nearly six years ago to go with my scheduler (before **uasyncio** was usable). I ported them to **uasyncio** when it grew legs. In all that time you are the first to request this feature. Unless further demand emerges I'm reluctant to add features/bloat - in fact my (long) todo list includes trying to make this class more lightweight.

For the moment I suggest you use your own fork.

**Comment by peterhinch** [comment]
I've now had a chance to review the Pushbuton class with reference to this PR and another proposed change (which I have not implemented). On reflection I think I was hasty in rejecting this PR which does make sense. Thank you.

**Comment by andydhobbs** [comment]
Thank you for accepting the change!

**Pull request merged**